### PR TITLE
New Bubble chart pull request 

### PR DESCRIPTION
**Pull request by mdkhanNielsen**
I have added a few features on the bubble chart
1. bubble chart x axis log axis
2. y axis log axis
3. legend
4. display chart color
5. no series chart
6. option to add cartesian grid
7. tooltip data
8. More options to place legends
9. 4 stories on the story book

### CODE REVIEW

**Review by simcha90 — APPROVED** [review]

File: stories/plugin-chart-waterfall/WaterfallChart.stories.tsx [hunk 1]
```diff
@@ -18,7 +18,7 @@
  */
 import React from 'react';
 import { D3_FORMAT_OPTIONS } from '@superset-ui/chart-controls';
-import { ChartProps, supersetTheme, ThemeProvider } from '@superset-ui/core';
+import { supersetTheme, ThemeProvider } from '@superset-ui/core';
```
Comment (amitmiran137) [inline stories/plugin-chart-waterfall/WaterfallChart.stories.tsx:21]: is this mandatory?

File: plugins/plugin-chart-bubble/src/BubbleChart.tsx [hunk 2]
```diff
@@ -150,38 +156,71 @@ const BubbleChart: FC<BubbleChartProps> = props => {
...
+  data.map(item => legendIds.includes(item[groupingField]) ? null: legendIds
+    .push(item[groupingField]) && legends.push({
+          value: item[groupingField],
+          type: disabledItems.includes(item[groupingField] as string) ? 'line' : 'square',
+          id: item[groupingField],
+          color: CategoricalColorNamespace.getScale(colorScheme)(item[groupingField]),
+        }),
+  );
+
+  legends = legends.slice(0, 18);
```
Comment (amitmiran137) [inline plugins/plugin-chart-bubble/src/BubbleChart.tsx:189]: why are we slicing here?

**Review by amitmiran137 — CHANGES_REQUESTED** [review]
a few inital comments 
also delete package-lock.,json

File: stories/plugin-chart-waterfall/WaterfallChart.stories.tsx (hunk 1 above)
Comment (mdkhanNielsen) [reply stories/plugin-chart-waterfall/WaterfallChart.stories.tsx:21]: this was in the waterfall chart (I didn't work on that). However, while linting i was getting a lint error for unused variable `ChartProps`, I just removed that one. Didn't explore whether supersetTheme or ThemeProvider is mandatory or not

File: plugins/plugin-chart-bubble/src/BubbleChart.tsx (hunk 2 above)
Comment (mdkhanNielsen) [reply plugins/plugin-chart-bubble/src/BubbleChart.tsx]: I added a new controls 'Max Number of Legends' so, its not hard coded any more. However, the default is 18

**Comment by mdkhanNielsen** [comment]
> a few inital comments also delete package-lock.,json

I deleted the package-lock and yarn-lock file

**Comment by jayakrishnankk** [comment]
@mdkhanNielsen , @amitmiran137 - is this ready to be merged ?

**Comment by jayakrishnankk** [comment]
@amitmiran137 - what was the issue that we discussed in our superset meeting about bubble chart ? @mdkhanNielsen is not aware of any color issue.

**Pull request merged**